src/scripts/main.js

```diff
@@ -1,3 +1,162 @@
 'use strict';
 
 // write code here
+// Notifications
+
+const pushNotification = (posTop, posRight, title, description, type) => {
+  const notification = document.createElement('div');
+
+  notification.setAttribute('data-qa', 'notification');
+
+  const notificationTitle = notification.appendChild(
+    document.createElement('h1'),
+  );
+  const notificationDescription = notification.appendChild(
+    document.createElement('p'),
+  );
+
+  notificationDescription.textContent = description;
+  notificationTitle.textContent = title;
+
+  notification.classList.add(type);
+  notification.style.top = `${posTop}px`;
+  notification.style.right = `${posRight}px`;
+
+  document.body.appendChild(notification);
+
+  setTimeout(() => {
+    notification.style.display = 'none';
+  }, 2000);
+};
+
+// Sorting
+
+const sortButtons = document.querySelectorAll('th');
+const tableBody = document.querySelector('tbody');
+let ascOrder = true;
+const rows = Array.from(tableBody.rows);
+
+function sortTable(cellIndex, sortType) {
+  rows.sort((rowA, rowB) => {
+    let cellA = rowA.cells[cellIndex].textContent.trim();
+    let cellB = rowB.cells[cellIndex].textContent.trim();
+
+    if (cellIndex === 3 || cellIndex === 4) {
+      cellA = parseFloat(cellA.replace(/[$,]/g, '')) || 0;
+      cellB = parseFloat(cellB.replace(/[$,]/g, '')) || 0;
+
+      if (sortType === 'ASC') {
+        return cellA - cellB;
+      } else {
+        return cellB - cellA;
+      }
+    }
+
+    if (sortType === 'ASC') {
+      return cellA.localeCompare(cellB);
+    } else {
+      return cellB.localeCompare(cellA);
+    }
+  });
+  rows.forEach((row) => tableBody.appendChild(row));
+}
+
+sortButtons.forEach((button) => {
+  button.addEventListener('click', () => {
+    const sortType = ascOrder ? 'ASC' : 'DESC';
+
+    sortTable(button.cellIndex, sortType);
+    ascOrder = !ascOrder;
+  });
+});
+
+// Selection
+
+rows.forEach((row) => {
+  row.addEventListener('click', () => {
+    row.classList.remove('active');
+
+    row.classList.add('active');
```

Comment:
The current logic removes the 'active' class and then immediately adds it back to the same row. This doesn't achieve the intended effect of toggling the 'active' state. Consider checking if the row already has the 'active' class and toggle it accordingly.